contrib/babelfishpg_tds/src/backend/tds/tdscomm.c

```diff
@@ -886,13 +930,27 @@ TdsReadNextRequest(StringInfo message, uint8_t *status, uint8_t *messageType)
 		if (TdsGetbytes(message->data + message->len, readBytes))
 			return EOF;
 		message->len += readBytes;
+
+		if (TdsRecvMessageType == TDS_BULK_LOAD)
+		{
+			/* if this is the last packet then notify the caller. */
+			if (TdsRecvPacketStatus & TDS_PACKET_HEADER_STATUS_EOM)
+				BulkCopyPacketStatus = 1;
+			TdsDoProcessHeader = true;
```

Comment:
if we remove this whole condition and enquire packet status separately, is this needed? wouldn't 946 will work for us

Reply:
No we will still need to return after reading the first packet only and NOT the entire request. If you see for line 946 to be executed it needs to be the last packet of that request